wordpress-mobile/WordPress-iOS
# iPad Splitview: Me

Introduces `UISplitViewController` to the Me section of the app.

As we've discussed previously, this tries to mimic the system Settings app as much as possible. So the left pane always shows the top level navigation, and the right pane can be used to navigate into each item. I've also made the left pane wider in landscape, to match Settings, and it's always visible in both portrait and landscape.

I've made a number of customisations to `UISplitViewController` to support the behaviour we require, in the subclass `WPSplitViewController`. For example, it handles wrapping and unwrapping child view controllers in navigation controllers where required. 

It also handles automatically populating the right pane with a detail view controller through a protocol `WPSplitViewControllerDetailProvider`. A primary view controller (left pane) can implement the method 

```
func initialDetailViewControllerForSplitView(splitView: WPSplitViewController) -> UIViewController?
```

And return a view controller that should populate the right pane on first appearance. For example, the Me section shows the My Profile detail screen.

To test: 

* Try the Me section out on both iPhone and iPad, moving between orientations and multitasking configurations. 
* Navigate into each item within Me, and try to follow every navigation route to ensure things transition as you'd expect.

Needs review: @kurzee 

We'll review this, and I'll start working on the next section. It's quite likely that `WPSplitViewController` will need some tweaks to support other sections of the app, so we can see where that takes us before we merge any of this.
